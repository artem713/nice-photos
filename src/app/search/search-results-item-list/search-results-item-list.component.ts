import { Component, Input } from '@angular/core';

@Component({
  selector: 'app-search-results-item-list',
  templateUrl: './search-results-item-list.component.html',
  styles: ['mat-grid-list {width: 90%; left: 5%;}']
})

export class SearchResultsItemListComponent {
  @Input() items; // TODO: add type after search component refactoring
}