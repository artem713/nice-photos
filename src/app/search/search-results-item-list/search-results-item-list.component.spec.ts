import { TestBed, ComponentFixture } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { MatGridListModule } from '@angular/material';
import { SearchResultsItemListComponent } from './search-results-item-list.component';

describe('SearchResultsItemListComponent', () => {
  let component: SearchResultsItemListComponent;
  let fixture: ComponentFixture<SearchResultsItemListComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [SearchResultsItemListComponent],
      imports: [MatGridListModule],
      schemas: [NO_ERRORS_SCHEMA]
    });

    fixture = TestBed.createComponent(SearchResultsItemListComponent);
    component = fixture.componentInstance;
  });

  it('should be rendered correctly', () => {
    component.items = [1, 2, 3];
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});