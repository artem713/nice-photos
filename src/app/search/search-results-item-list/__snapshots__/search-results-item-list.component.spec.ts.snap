// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsItemListComponent should be rendered correctly 1`] = `
<app-search-results-item-list
  items={[Function Array]}
>
  <mat-grid-list
    class="mat-grid-list"
    cols="3"
    guttersize="15px"
    ng-reflect-cols="3"
    ng-reflect-gutter-size="15px"
    ng-reflect-row-height="15em"
    rowheight="15em"
  >
    <div>
      
    
      
      <mat-grid-tile
        class="mat-grid-tile"
        style="left: 0px; top: 0px;"
      >
        <figure
          class="mat-figure"
        >
          
      
          <app-search-results-item />
          
    
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
        style="top: 0px;"
      >
        <figure
          class="mat-figure"
        >
          
      
          <app-search-results-item />
          
    
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
        style="top: 0px;"
      >
        <figure
          class="mat-figure"
        >
          
      
          <app-search-results-item />
          
    
        </figure>
      </mat-grid-tile>
      
  
    </div>
  </mat-grid-list>
</app-search-results-item-list>
`;
